astropy/coordinates/sky_coordinate.py

```diff
@@ -1225,6 +1225,92 @@ def from_pixel(cls, xp, yp, wcs, origin=0, mode='all'):
+        obstime : `~astropy.time.Time` or None, optional
+            The time at which to compute the correction.  If `None`, the
+            ``obstime`` frame attribute on the `SkyCoord` will be used. (Raises
+            an exception if the `SkyCoord` and this argument are in conflict.)
+        location : `~astropy.coordinates.EarthLocation` or None, optional
+            The observer location at which to compute the correction.  If
+            `None`, the  ``location`` frame attribute on the `SkyCoord` will be
+            used. (Raises an exception if the `SkyCoord` and this argument are
+            in conflict)
+
+
+        Returns
+        -------
+        vcorr : astropy.units.Quantity with velocity units
+            The  correction with a positive sign.  I.e., *add* this
+            to an observed radial velocity to get the heliocentric velocity.
+
+        Notes
+        -----
+        The algorithm here is sufficient to perform corrections at the ~1 to
+        10 m/s level, but has not been validated at higher prevision.  Future
+        versions of Astropy will likely aim to improve this.
+
+        The default is for this method to use the builtin ephemerides for
+        computing the sun and earth location.  This can be changed by setting the
+        `~astropy.coordinates.solar_system`_ variable, either directly or via
+        ``with`` statement.
+        """
+        # has to be here to prevent circular imports
+        from .solar_system import get_body_barycentric_posvel
+
+        t = self.obstime
+        if (obstime is not None and self.obstime is not None and
+           obstime != self.obstime):
+            raise TypeError('cannot compute radial velocity correction if '
+                            'obstime frame argument is passed in and it is '
+                            'inconsistent with the obstime on the SkyCoord')
+        elif self.obstime is None:
+            t = obstime
+        if t is None:
+            raise ValueError('Must provide an `obstime` to '
+                             'radial_velocity_correction, either as a SkyCoord'
+                             ' frame attribute or in the method call.')
+
+        loc = self.location
+        if (location is not None and self.location is not None and
+           location != self.location):
+            raise TypeError('cannot compute radial velocity correction if '
+                            'location frame argument is passed in and it is '
+                            'inconsistent with the location on the SkyCoord')
+        elif self.location is None:
+            loc = location
+        if loc is None:
+            raise ValueError('Must provide a `location` to '
+                             'radial_velocity_correction, either as a SkyCoord'
+                             ' frame attribute or in the method call.')
+
+        if kind == 'barycentric':
+            vorigintoearth = get_body_barycentric_posvel('earth', t)[1]
```

Comment:
I would use underscores for any of these velocity variables `v_origin_to_earth` and `v_sun` etc. But maybe that's a style thing...